docs/runtime/core.md

```diff
@@ -1 +1,73 @@
 # Core
+
+Runtime Core is a compilation/execution engine for NN models.
+
+## Modules
+
+Runtime Core has four modules. These are namespaces as well as directory names.
+
+- `ir`  stands for Intermediate Representation which contains Neural Network Graph data structures
+- `compiler` converts IR to execution, backends can be used for compilation
+- `exec` is execution module which is the result of compilation.
+- `backend` is backend interface
+
+### Module `ir`
+
+This module contains data structures of pure Neural Network models. The models from NN Packages or NN API are converted to these structures.
+
+- `Subgraphs` is the entire neural network model which is a group of subgraphs
+- `Subgraph` consists of operands and operations
+- `Operand` (a.k.a. Tensor) has shape, data type, data and references to operations
+- `Operation` (a.k.a. Operator) has operation type, params and references to operands
+
+`Operand` and `Operation` are graph nodes. References to operations and operands are graph edges.
+
+![Core](core-figure-ir.png)
+
+`Subgraphs` represents the whole model. It could have more than one `Subgraph` to support control flow operations. Those operations can make calls to another subgraph and when the execution on another subgraph is done it gets back to previous subgraph execution with returned operands.
+
+All `Graph`s are a [DAG](https://en.wikipedia.org/wiki/Directed_acyclic_graph) so once model inputs are given we can run it in topological order.
+
+### Module `compiler`
+
+`Compiler` is the main class of this module. Everything starts from it.
+
+What it does is making models executable. It schedules run order and assigns a backend for each operation. Here are major tasks of compilation.
+
+#### 1. Lowering
+
+In Lowering, the compiler assigns a [backend](#) for each operation. A backend is assigned to an operation means that that operation will be run with the assigned backend's kernel.
+
+There is a scheduler that allows the user manually specify backends via compiler options. There is another scheduler that automatically assigns backends based on profiling info.
+
+#### 2. Tensor Registration
+
+Each backend manages its tensors. In this phase operand informations get registered to the corresponding backend. This will be used generating tensor objects.
```

Comment:
> ... In this phase ...

We haven't mentioned `pahse` before. Wouldn't it be nice if you briefly explained it earlier? Anyway, it's an option. :)

Reply:
I changed "task" to "phase".